File: Arcade/Intro/17.clj
(defn arrayChange [inputArray]
  "arrayChange : [Int] -> Int.
   Increase by one the elements inside inputArray until its
   increasing array.
   By diko1"
  (loop [[first_ second_ & rest_] inputArray
         moves 0]
    (cond
      (nil? second_) moves
      (< first_ second_) (recur (cons second_ rest_) moves)
      :otherwise (let [diff (inc (- first_ second_))]
                   (recur (cons  (inc first_) rest_) (+ moves diff))))))

File: Arcade/Intro/17.cpp
int arrayChange(std::vector<int> inputArray) {
  // arraychange : [Int] -> Int.
  // Increase by one the elements inside inputArray until its
  // increasing array.
  int acumulator = 0;
  
  for(int i = 1; i < inputArray.size; i++) {
    int tmpSub = inputArray[i - 1] - inputArray[i];
    if(tmpSub >= 0) {
      inputArray[i] += tmpSub + 1;
      acumulator += tmpSub + 1;
    }
  }
  
  return acumulator;
}
